api/src/main/java/com/velocitypowered/api/plugin/PluginContainer.java

```diff
@@ -0,0 +1,26 @@
+package com.velocitypowered.api.plugin;
+
+import java.util.Optional;
+import java.util.logging.Logger;
+
+/**
+ * A wrapper around a class marked with an {@link Plugin} annotation to
+ * retrieve information from the annotation for easier use.
+ */
+public interface PluginContainer extends PluginDescription {
+    /**
+     * Returns the created instance of {@link Plugin} if it is available.
+     *
+     * @return the instance if available
+     */
+    default Optional<?> getInstance() {
+        return Optional.empty();
+    }
+
+    /**
+     * Returns the assigned logger to this {@link Plugin}.
+     *
+     * @return the assigned logger
+     */
+    Logger getLogger();
```

Comment:
This should use a slf4j logger instead. JUL doesn't work well on top of Log4J 2.

Reply:
I'm gonna need to include Log4J 2 on the API module then (check the TODO on the implementation)

Reply:
It already uses log4j2, so all that is needed is `compile 'org.slf4j:slf4j-api:1.7.25'` in the API and `compile "org.apache.logging.log4j:log4j-slf4j-impl:${log4jVersion}"` in the proxy.

Reply:
No, you won't need to do that. Just include the SLF4J API in the API module and then the log4j connector on the proxy.